### ISSUE TITLE
Share files with correct path and not into folder "talk"

### ISSUE THREAD

**Issue by pschopen** [nextcloud/spreed#8950]
We have a group with all team members and a common folder with all commonly used files that is shared to that group. Whenever one attaches a file of that shared folder to a card in deck and one team member clicks on that card it is opened showing inside the common folder.

However, whenever someone shares such a file in talk and then opens it, it points to the "talk" folder instead of the common shared folder. When you then delete this file from the talk folder (Freigabe verlassen), go back to talk and open it then, it points you to the correct place inside the shared folder.

For me it looks like talk always "shares" a shared file directly to everyone in the chat without checking in advance if the folder the file is from is already shared to the members of the chat.

For working with files it sometimes make sense to reveal the folder structure as well so that you can see the context of that file. So it would be good to implement a check if the file is already shared to the members of the chat before sharing it again.

**Comment by pschopen** [comment]
I investigated the issue further. Indeed, Deck also shares the file to the deck-folder. However, the order when opening seems to be different. Deck first tries to open the file via the commonly shared folder and only uses the "deck"-folder as a backup while talk does it the other way around. It tries to open the file from the "talk"-folder first and only if it's missing there finds it inside the commonly shared folder. Behaviour in talk should be the same as in deck in my opinion.

**Comment by nickvergessen** [comment]
Right, we populate the share data instead of making another DB request:
https://github.com/nextcloud/spreed/blob/master/lib/Chat/Parser/SystemMessage.php#L658-L664

**Comment by pschopen** [comment]
I see that the decision makes sense for performance reasons. For the workflow I think it's problematic as the context of the file isn't told. You e.g. don't know if there is a _v2-Version of the file or how it relates to the folder structure and the other files. I actually think that this would be worth another DB request.
